packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -2676,6 +2676,35 @@ class UnconstrainedBox extends StatelessWidget {
 ///
 /// {@youtube 560 315 https://www.youtube.com/watch?v=PEsY654EGZ0}
 ///
+/// {@tool dartpad --template=stateless_widget_scaffold}
+///
+/// This sample shows [FractionallySizedBox] widget that has one container
+/// with 50% height and width of the available space of parent widget.
+///
+/// If [Alignment] property is specified, then [FractionallySizedBox]
+/// should be Constraining.
+///
+/// ```dart
+/// Widget build(BuildContext context) {
```

Comment:
This looks good, although we no longer use the `new` keyword. I replaced Container with DecoratedBox, since that's sufficient, and used `4` instead of `4.0` since that's become conventional.

```dart
Widget build(BuildContext context) {
  return SizedBox.expand(
    child: FractionallySizedBox(
      widthFactor: 0.5,
      heightFactor: 0.5,
      alignment: FractionalOffset.center,
      child: DecoratedBox(
        decoration: BoxDecoration(
          border: new Border.all(
            color: Colors.blue,
            width: 4,
          ),
        ),
      ),
    ),
  );
}
```